### PR TITLE
Fix examples and templates compiling on GNU/Linux

### PR DESCRIPTION
**Pull request by LelixSuper**
For GNU/Linux I've clean the compiling command and added support for standard directories.
For the total project, I've changed default targets names to 'all', I've followed the [GNU guidelines](https://www.gnu.org/prep/standards/standards.html#Standard-Targets).

### CODE REVIEW

**Comment by LordRibblesdale** [comment]
+1

**Comment by raysan5** [comment]
Hi @LelixSuper! Many thanks for the changes. 

I think you are using tabs instead of spaces. Please, could you change tabs by 4 spaces? I'm following that convention in raylib.

If it suppose a lot of trouble I can integrate your pull-request and update it later. Let me know your preference.

**Comment by LelixSuper** [comment]
@raysan5 I've converted all tabs into spaces manually. I've only a question: on GNU/Linux (such as Ubuntu), you need always to uncomment the 56th line for compiling a template or an example. Why is it commented for default?

**Comment by raysan5** [comment]
Excuse me @LelixSuper, what line are you referring to? Maybe it can be uncommented...

File: examples/Makefile [hunk 1]
```diff
@@ -114,9 +148,9 @@ ifeq ($(PLATFORM),PLATFORM_DESKTOP)
         # libraries for Debian GNU/Linux desktop compiling
         # requires the following packages:
         # libglfw3-dev libopenal-dev libglew-dev libegl1-mesa-dev
-        LIBS = -lraylib -lglfw3 -lGLEW -lGL -lopenal -lm -pthread
-	# on XWindow could require also below libraries, just uncomment
-	#LIBS += -lX11 -lXrandr -lXinerama -lXi -lXxf86vm -lXcursor
+        LIBS = -lraylib -lglfw3 -lGLEW -lGL -lopenal -lm -pthread -ldl
+        # on XWindow could require also below libraries, just uncomment
+        #LIBS += -lX11 -lXrandr -lXinerama -lXi -lXxf86vm -lXcursor
```
Comment (LelixSuper) [inline examples/Makefile]: This line @raysan5 (obviously also other Makefiles).

**Comment by raysan5** [comment]
Ok, I see, I thought it was not required (I don't need those dependencies on Raspberry Pi raylib version). Those dependencies probably come from GLFW3. It could be enabled by default on PLATFORM=PLATFORM_DESKTOP and PLATFORM_OS=LINUX.

By the way, I see that GLEW library is also listed for linking, -lGLEW could be removed, raylib 1.4 replaced it by GLAD loading library (probably it supposed the -ldl dependency).

**Comment by LelixSuper** [comment]
@raysan5 I'm speaking about line 153, if you want to compile on GNU/Linux you must uncomment the line. Thus, can you remove GLEW definitively and uncomment the 153th line?

**Comment by raysan5** [comment]
Yes, you can.

**Comment by LelixSuper** [comment]
@raysan5 I've finished, now you can merge :smile:

**Comment by raysan5** [comment]
Merged! :D 

Thank you very much @LelixSuper for your help and for your time!

**Comment by LelixSuper** [comment]
You're welcome!
If you want, you can ad me in "acknowledgements" section, but you do not feel obligated!

**Pull request merged**